### PR TITLE
chore(processors.regex): Inform and warn user on operation modus

### LINKED ISSUE #14028: processor.regex mixing non capture groups and named capture groups gives incorrect result
### Relevant telegraf.conf

```toml
[[processors.regex]]
  order = 20
  namepass = ["arista"]
  metricpass = "fields.process == 'Bgp'"

  [[processors.regex.fields]]
    # extract peer
    key = "message"
    #pattern = '.*(peer|neighbor) (?P<ip_addr>\d+\.\d+\.\d+\.\d+).*'     # this does not work
    pattern = '.*neighbor (?P<ip_addr>\d+\.\d+\.\d+\.\d+).*'    # this works
```

### Logs from Telegraf

```text
N/A
```

### System info

Telegraf nightly build

### Docker

_No response_

### Steps to reproduce

Using nightly build to test https://github.com/influxdata/telegraf/pull/13971 I get unexpected results when mixing non capture and named capture groups.

Using this input line:

received from neighbor 1.1.1.1 (VRF default AS 4200099061) 6/5 (Cease/connection rejected) 0 bytes

And telegraf.conf using regex pattern '.*(peer|neighbor) (?P<ip_addr>\d+\.\d+\.\d+\.\d+).*' is get this:

arista process="Bgp",event="BGP-3-NOTIFICATION",message="",source="lab" 1696045222031424000

The message is replaced with "". Note the above regex works correctly in regex101 tester.

### Expected behavior

Should result in:

arista source="lab",process="Bgp",event="BGP-3-NOTIFICATION",message="received from neighbor 1.1.1.1 (VRF default AS 4200099061) 6/5 (Cease/connection rejected) 0 bytes",ip_addr="1.1.1.1",vrf="default",resource_index="4200099061" 1696045740257530000

### Actual behavior

Orignal source field is replaced with "".

### Additional info

Possibly an additional code check is required to check if the 'replacement = ' config option is defined. If not, assume named capture groups used and don't fall through into replacement mode with empty key.

### PR DESCRIPTION
**Pull request by srebhan**
- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #14028

This PR adds an information log-message on the used modus (explicit vs. named-groups) and warns in case no explicit options are given but unnamed or mixed (named & unnamed) groups are used.

### CODE REVIEW

**Review by Hipska — APPROVED** [review]

**Comment by richardeaxon** [comment]
Awesome thanks. I downloaded and tested against tiger build and it works perfectly:

```
2023-10-11T22:37:40Z I! Starting Telegraf 1.29.0-a2ad4d05 brought to you by InfluxData the makers of InfluxDB
2023-10-11T22:37:40Z I! Available plugins: 241 inputs, 9 aggregators, 29 processors, 24 parsers, 60 outputs, 5 secret-stores
2023-10-11T22:37:40Z I! Loaded inputs: tail
2023-10-11T22:37:40Z I! Loaded aggregators:
2023-10-11T22:37:40Z I! Loaded processors: defaults lookup (3x) regex (3x) strings (2x)
2023-10-11T22:37:40Z I! Loaded secretstores:
2023-10-11T22:37:40Z I! Loaded outputs: file
2023-10-11T22:37:40Z I! Tags enabled: host=collector1
2023-10-11T22:37:40Z I! [agent] Config: Interval:30s, Quiet:false, Hostname:"collector1", Flush Interval:10s
2023-10-11T22:37:40Z D! [agent] Initializing plugins
2023-10-11T22:37:40Z I! [processors.regex] fields: Using explicit mode...
2023-10-11T22:37:40Z I! [processors.regex] fields: Using explicit mode...
2023-10-11T22:37:40Z I! [processors.regex] fields: Using explicit mode...
2023-10-11T22:37:40Z I! [processors.regex] fields: Using explicit mode...
2023-10-11T22:37:40Z I! [processors.regex] fields: Using explicit mode...
2023-10-11T22:37:40Z I! [processors.regex] fields: Using explicit mode...
2023-10-11T22:37:40Z I! [processors.regex] fields: Using explicit mode...
2023-10-11T22:37:40Z W! [processors.regex] fields: Neither 'result_key' nor 'replacement' given with unnamed or mixed groups.
Using explicit, empty replacement!
2023-10-11T22:37:40Z I! [processors.regex] fields: Using named-group mode...
2023-10-11T22:37:40Z I! [processors.regex] fields: Using named-group mode...

```

File: plugins/processors/regex/converter.go [hunk 1]
```diff
@@ -39,8 +78,15 @@ func (c *converter) setup(ct converterType) error {
 				}
 			}
 			if allNamed {
+				log.Infof("%s: Using named-group mode...", ct)
 				c.groups = groups[1:]
+			} else {
+				msg := "Neither 'result_key' nor 'replacement' given with unnamed or mixed groups.\n"
+				msg += "Using explicit, empty replacement!"
```
Comment (powersj) [inline plugins/processors/regex/converter.go:85]: I don't believe we have a history of doing multi-line log messages. Could this either be on a single line, two separate warnings, or an info + warning?

Comment (Hipska) [reply plugins/processors/regex/converter.go:85]: I had my doubts on this one too.

File: plugins/processors/regex/converter.go [hunk 2]
```diff
@@ -39,8 +78,15 @@ func (c *converter) setup(ct converterType) error {
 				}
 			}
 			if allNamed {
+				log.Infof("%s: Using named-group mode...", ct)
 				c.groups = groups[1:]
+			} else {
+				msg := "Neither 'result_key' nor 'replacement' given with unnamed or mixed groups;"
+				msg += "using explicit, empty replacement!"
```
Comment (powersj) [inline plugins/processors/regex/converter.go:85]:
```suggestion
				msg += " using explicit, empty replacement!"
```

**Review by powersj — APPROVED** [review]

**Pull request merged**